Models/Config.cs

```diff
@@ -0,0 +1,126 @@
+using Miki.Logging;
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Threading.Tasks;
+
+namespace Miki.Bot.Models
+{
+    [Table("Configuration")]
+    public class Config
+    {
+        [Key]
+        [Column("Id")]
+        public Guid Id { get; set; } = new Guid();
+
+        /// <summary>
+        /// Discord API Token
+        /// </summary>
+        [Column("Token")]
+        public string Token { get; set; } = "";
+
+        /// <summary>
+        /// Sentry Error Tracking
+        /// </summary>
+        [Column("SharpRavenKey")]
+        public string SharpRavenKey { get; set; } = "";
+
+        /// <summary>
+        /// Datadog Agent host
+        /// </summary>
+        [Column("DatadogHost")]
+        public string DatadogHost { get; set; } = "127.0.0.1";
+
+        /// <summary>
+        /// Cache connection string
+        /// </summary>
+        [Column("RedisConnectionString")]
+        public string RedisConnectionString { get; set; } = "localhost";
+
+        /// <summary>
+        /// Miki API route
+        /// </summary>
+        [Column("MikiApiBaseUrl")]
+        public string MikiApiBaseUrl { get; set; } = "https://api.miki.ai/";
+
+        /// <summary>
+        /// Miki API Key
+        /// </summary>
+        [Column("MikiApiKey")]
+        public string MikiApiKey { get; set; } = "";
+
+        /// <summary>
+        /// Image API route
+        /// </summary>
+        [Column("ImageApiUrl")]
+        public string ImageApiUrl { get; internal set; } = "";
+
+        [Column("CdnRegionEndpoint")]
+        public string CdnRegionEndpoint { get; internal set; } = "";
+
+        [Column("CdnAccessKey")]
+        public string CdnAccessKey { get; internal set; } = "";
+
+        [Column("CdnSecretKey")]
+        public string CdnSecretKey { get; internal set; } = "";
+        
+        [Column("RabbitUrl")]
+        public string RabbitUrl { get; internal set; } = "amqp://localhost";
+        
+        [Column("DanbooruCredentials")]
+        public string DanbooruCredentials { get; internal set; } = "";
+        
+        [Column("BunnyCdnKey")]
+        public string BunnyCdnKey { get; internal set; }
+        public static async Task<Config> InsertNewConfigAsync(string connStr)
+        {
+            var builder = new DbContextOptionsBuilder<MikiDbContext>();
+            builder.UseNpgsql(connStr, b => b.MigrationsAssembly("Miki.Bot.Models"));
+            var dbContext = new MikiDbContext(builder.Options);
+
+            var configuration = new Config();
+
+            await dbContext.Configurations.AddAsync(configuration);
+
+            await dbContext.SaveChangesAsync();
+
+            Log.Debug("New Config inserted into database with Id: " + configuration.Id);
+
+            return configuration;
+        }
+
+        public static async Task<Config> GetOrInsertAsync(string connStr, string configId = null)
```

Comment:
Try to keep methods on an alphabetic order

Reply:
Would recommend making the default Guid simply Guid(0) so you can query stateless